test/test_null_layout.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <numeric>
+
+#include "sparrow/array_data_factory.hpp"
+#include "sparrow/null_layout.hpp"
+
+#include "doctest/doctest.h"
+
+namespace sparrow
+{
+    TEST_SUITE("null layout")
+    {
+        TEST_CASE("constructor")
+        {
+            constexpr std::size_t size = 5u;
+            array_data ad = make_array_data_for_null_layout(size);
+            null_layout nl(ad);
+            CHECK_EQ(nl.size(), size);
+        }
+
+        TEST_CASE("rebind_data")
+        {
+            constexpr std::size_t size1 = 5u;
+            constexpr std::size_t size2 = 5u;
```

Comment:
you should use a different size to be sure that the rebind is ok

Reply:
omg I thought I did it ...